src/dynamics/video_player/player/player.js

```diff
@@ -2995,6 +2997,11 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                     this.set("adsplayer_active", false);
                 },
 
+                fullscreenElement: function() {
+                    if (this.get("shadow")) return document.querySelector("#player");
```

Comment:
Hi @Josewd, the player doesn't always have id `player`, so we might want to try `this.activeElement().host` instead. 🙂

There are also cases where the active element isn't inside a shadow DOM due to `shadow` attribute, but because the element that was passed to the constructor was already inside one (see demos/shadow-dom.html on other repo). To address those cases we could try using `this.activeElement().toString() === "[object ShadowRoot]"` instead of `this.get("shadow")`

Reply:
Hi, @ntshcalleia 
I tried the `this.activeElement().host` but it just returns undefined. so the other workaround would be using closure to check if shadowRoot is in fullscreen or not. because I could not find another way to select the parent of parent of shadowRoot which is the element being checked on document.fullscreenElement on Beta.js method.